modules/sfp_countryname.py

```diff
@@ -0,0 +1,308 @@
+            matchCountries = re.findall("[\s,'\"]" + countryName + "[\s,'\"]", srcWhoIs, re.IGNORECASE)
+
+            if len(matchCountries) > 0:
+                # Get country name from first index of list
+                # Extract only the text part of the country code
+                matchCountry = matchCountries[0].strip(",").strip("'").strip("\"").strip()
+                countries.append(matchCountry)
+        
+        # If any countries are found
+        if len(countries) > 0:
+            return countries
+
+        return None
+            
+    # What events is this module interested in for input
+    def watchedEvents(self):
+        return ["IBAN_NUMBER", "PHONE_NUMBER", "AFFILIATE_DOMAIN_NAME",
+                "CO_HOSTED_SITE_DOMAIN", "DOMAIN_NAME", "SIMILARDOMAIN",
+                "AFFILIATE_DOMAIN_WHOIS", "CO_HOSTED_SITE_DOMAIN_WHOIS",
+                "DOMAIN_WHOIS", "GEOINFO", "PHYSICAL_ADDRESS"]
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["COUNTRY_NAME"]
+
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+        
+        evttype = "COUNTRY_NAME"
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        myres = list()
+
+        # Generate event data hash
+        eventDataHash = self.sf.hashstring(eventData)
+        # Don't parse duplicate data
+        if eventDataHash in myres:
+            self.sf.debug("Already found from this source")
+            return None 
+        
+        myres.append(eventDataHash)
+
+        countryNames = list()
+
+        # Process the event data based on incoming event type
+        if eventName == "PHONE_NUMBER":
+            countryNames.append(self.detectCountryFromPhone(eventData))
+        elif eventName == "DOMAIN_NAME" or (eventName == "AFFILIATE_DOMAIN_NAME" and self.opts["affiliate"]) or(eventName == "CO_HOSTED_SITE_DOMAIN" and self.opts["cohosted"]) or  (eventName == "SIMILARDOMAIN" and self.opts["similardomain"]):
+            countryNames.append(self.detectCountryFromTLD(eventData))
+        elif eventName == "IBAN_NUMBER":
+            countryNames.append(self.detectCountryFromIBAN(eventData))
+        elif eventName in ["DOMAIN_WHOIS", "GEOINFO", "PHYSICAL_ADDRESS"] or (eventName == "AFFILIATE_DOMAIN_WHOIS" and self.opts["affiliate"]) or (eventName == "CO_HOSTED_SITE_DOMAIN_WHOIS" and self.opts["cohosted"]):
+            countryNames.extend(self.detectCountryFromData(eventData))
+        
+        if None in countryNames:
+            countryNames.remove(None)
+        if len(countryNames) == 0 :
+            return None
+
+        # Remove duplicates if there are any
+        countryNames = set(countryNames)
+
+        for countryName in countryNames:
+            self.sf.debug("Found country name : " + countryName)
+            myres.append(countryName)
```

Comment:
You are duplicating the use of myres.. in the first case you're trying to use it to avoid processing duplicate events (which is fine), but here you're also using it to avoid /producing/ duplicate events, and that will cause problems. I suggest two separate checks - the one above to avoid handling the same events, and one here which only uses a variable scoped to this function to avoid producing multiple country names from the same source event.